services/user_service.py

```diff
@@ -35,9 +37,37 @@ def update_user(self, email: str, updated_data: dict) -> Optional[User]:
 
     def get_user_details(self, email: str) -> Optional[User]:
         return self.user_repository.get_user_by_email(email)
-
+    
     def remove_user(self, email: str):
         user = self.user_repository.get_user_by_email(email)
         if user is None:
             raise ValueError("User does not exist.")
-        return self.user_repository.delete_user(email)
\ No newline at end of file
+        return self.user_repository.delete_user(email)
+
+    def generate_profile_picture_path(self, email:str, image: UploadFile ):
+        user = self.user_repository.get_user_by_email(email)
+        if user is None:
+            raise ValueError("User does not exist.")
+        file_path = f"uploads/profile_pictures/{email}/{image.filename}" 
+        os.makedirs(os.path.dirname(file_path), exist_ok=True)
+        with open(file_path, "wb") as buffer:
+            shutil.copyfileobj(image.file, buffer)
+        return file_path
+    
+    def delete_profile_picture(self, email:str):
```

Comment:
I don't think we should have a delete profile picture method, but instead a method to update the profile picture, because by default a user with no profile picture will still have a default picture, so in case the user wants to delete his picture we just update it with the default one. what do you think @Renwarren @bjoelfra ?

Reply:
But if we just reset the default value the picture is still in the database we have just removed the path associated to the user

Reply:
Definitely. Agree with this.